### PR TITLE
Add CollisionService

### PR DESCRIPTION
**Pull request by Nicklason**
Add collision service implemented using grids.

Splits the entire game map into grids and detects collision inside of each grid.

### CODE REVIEW

**Review by Kjalris — APPROVED** [review]

**Review by DenBedste — APPROVED** [review]

File: dk.sdu.se4.bullet/src/main/java/dk/sdu/se4/bullet/Bullet.java [hunk 1]
```diff
@@ -12,5 +12,7 @@
  * @author Kasper
  */
 class Bullet extends Entity{
-    
+    public Bullet() {
+        super(true);
```
Comment (stped18) [inline dk.sdu.se4.bullet/src/main/java/dk/sdu/se4/bullet/Bullet.java:16]: what is this
use an part insted

File: dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/entity/Entity.java [hunk 2]
```diff
@@ -20,11 +21,16 @@ public String getId() {
     }
 
     
-    public Entity() {
+    public Entity(boolean friendly) {
```
Comment (stped18) [inline dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/entity/Entity.java:24]: User an part insted

File: dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/entity/Entity.java [hunk 3]
```diff
@@ -20,11 +21,16 @@ public String getId() {
...
         log.info("Created {}",this.getClass().getName());
         this.id = UUID.randomUUID().toString();
         this.partStorage = new ConcurrentHashMap<>();
         
+        this.friendly = friendly;
+    }
+    
+    public boolean isFriendly() {
```
Comment (stped18) [inline dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/entity/Entity.java:32]: create TypePart and use that :) the project is based on parts

**Review by stped18 — CHANGES_REQUESTED** [review]

File: dk.sdu.se4.bullet/src/main/java/dk/sdu/se4/bullet/Bullet.java (hunk 1 above)
Comment (Nicklason) [reply dk.sdu.se4.bullet/src/main/java/dk/sdu/se4/bullet/Bullet.java:16]: Yeah my bad

**Comment by Nicklason** [comment]
I don't know why but this pull request was made as a draft because we were not done working, but it is now open for some reason. I will close it and make a new pull request when it is ready to merge.

**Pull request closed without merging**